roles/ocp_on_libvirt/tasks/libvirt_host_up.yml

```diff
@@ -1,74 +1,124 @@
-  ansible.builtin.command: "qemu-img resize {{ libvirt_image_path }}/{{ host['name'] }}.qcow2 +{{ host['disk_size'] }}"
   become: "{{ libvirt_become }}"
-  when: host['disk_size'] is defined and ool_node_not_exists
+  ansible.builtin.command: "qemu-img resize {{ libvirt_image_path }}/{{ host['name'] }}.qcow2 +{{ host['disk_size'] }}"
+  loop: "{{ hosts }}"
+  loop_control:
+    loop_var: host
+  when: host['disk_size'] is defined and host['name'] in ool_nodes_list
 
 - name: "Create an additional disk"
-  ansible.builtin.command: "qemu-img create -f qcow2 {{ libvirt_image_path }}/{{ host['name'] }}-disk2.qcow2 {{ host['disk2_size'] }} "
+  ansible.builtin.command: "qemu-img create -f qcow2 {{ libvirt_image_path }}/{{ host['name'] }}-disk2.qcow2 {{ host['disk2_size'] }}"
   become: "{{ libvirt_become }}"
+  loop: "{{ hosts }}"
+  loop_control:
+    loop_var: host
   when:
-    - host['disk2_size'] is defined and ool_node_not_exists
+    - host['disk_size'] is defined and host['name'] in ool_nodes_list
     - '"worker" in host["name"]'
 
 - name: "Create directories"
   ansible.builtin.file:
-    path: "/tmp/vm-{{ host['name'] }}"
+    path: "/tmp/vm-{{ item }}"
     state: "directory"
     mode: "0755"
-  when: ool_node_not_exists
+  loop: "{{ ool_nodes_list }}"
 
 - name: "Prepare cloud-config/user-data-local"
   ansible.builtin.template:
     src: "templates/cloud-config/user-data"
     dest: "/tmp/vm-{{ host['name'] }}/user-data"
     mode: "0644"
-  when: ool_node_not_exists
+  loop: "{{ hosts }}"
+  loop_control:
+    loop_var: host
+  when: host['name'] in ool_nodes_list
 
 - name: "Prepare cloud-config/meta-data"
   ansible.builtin.template:
     src: "templates/cloud-config/meta-data"
     dest: "/tmp/vm-{{ host['name'] }}/meta-data"
     mode: "0644"
-  when: ool_node_not_exists
+  loop: "{{ hosts }}"
+  loop_control:
+    loop_var: host
+  when: host['name'] in ool_nodes_list
 
 - name: "Generate ci data cd image for cloud-init when cloud config is defined"
   ansible.builtin.command: |
-    mkisofs -o /tmp/vm-{{ host['name'] }}.iso -V cidata -r -J
-      --quiet /tmp/vm-{{ host['name'] }}/user-data
-      /tmp/vm-{{ host['name'] }}/meta-data
+    mkisofs -o /tmp/vm-{{ item }}.iso -V cidata -r -J
+      --quiet /tmp/vm-{{ item }}/user-data
+      /tmp/vm-{{ item }}/meta-data
   become: "{{ libvirt_become }}"
-  when: ool_node_not_exists
+  loop: "{{ ool_nodes_list }}"
 
 - name: "Define node"
   community.libvirt.virt:
     name: "{{ host['name'] }}"
     command: define
     xml: "{{ lookup('template', 'libvirt_node.xml.j2') }}"
-  when: ool_node_not_exists
+  loop: "{{ hosts }}"
+  loop_control:
+    loop_var: host
+  when: host['name'] in ool_nodes_list
 
 - name: Sync memory data to disk
   ansible.builtin.command: sync
 
 - name: "Start VM"
   community.libvirt.virt:
-    name: "{{ host['name'] }}"
+    name: "{{ item }}"
+    state: "running"
+  loop: "{{ ool_nodes_list }}"
+
+- name: "Wait for VM to shut down"
```

Comment:
it is not clear to me how the VM goes from running to shutdown without intervention, I assume this expects the installer to shut it down? clarify with a comment in the code